### ISSUE TITLE
Flakiness in `RootViewSpec` + `RootPresenterSpec`

### ISSUE THREAD

**Issue by sashei** [mozilla-lockbox/lockbox-ios#768]
This is gathering from a conversation I had with @joeyg :

The `RootViewSpec` and `RootPresenterSpec` are the culprits for a lot of our flakiness in BuddyBuild with Lockbox for iOS. Adding tests to these suites substantially increases the risk of test failure / not getting test builds out, but as they are central infrastructure for the app we can't justify suspending them.

This story will constitute exploration / possible implementation of ways to better isolate these tests and prevent future difficulties adding new `View`s to the app.

**Comment by sashei** [comment]
Some thoughts that I've had:
We spin up a lot of resource-intensive `UIKit` classes in _both_ of these specs, more so in the `RootView`. We do not mock construction of any of the `View`s, which in addition to their expensive instantiation proceed to spin up `Presenter`s, which make calls to `Store`s.... regardless of if this passes locally, there's a _lot_ going on that's external to the tests. It might be helpful to implement better way of retrieving these views in the `RootView` (opportunity to use some type of `ViewFactory`?).

**Comment by joeyg** [comment]
@sashei I looked into creating a `ViewFactory` but got stuck with `RootView.topViewIs` being used to test the class type. Do you think that should use the view factory as well? https://github.com/joeyg/lockbox-ios/commit/a3f6f311fe0052771fc9141143bff5abb1a855f9

**Comment by sashei** [comment]
@joeyg yes, in my mind we would ask the mock of the `ViewFactory` what arguments it had been given rather than checking using the `topViewIs` method.
